parser/Node.h

```diff
@@ -16,6 +16,7 @@ class Node {
         return toStringWithTabs(gs);
     }
     virtual std::string toJSON(const core::GlobalState &gs, int tabs = 0) = 0;
+    virtual std::string toJSONWithLocs(const core::GlobalState &gs, core::FileRef file, int tabs = 0) = 0;
```

Comment:
My first try was merged with `toJSON` but it made the implementation less clear, it seemed better to make two different methods.